packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -654,6 +670,15 @@ export const Table = <
     setPageSize(pageSize);
   }, [pageSize, setPageSize]);
 
+  React.useEffect(() => {
+    if (previousFilter.current === currentFilter.current) {
+      return;
+    } else {
+      previousFilter.current = currentFilter.current;
+      onFilter?.(currentFilter.current, state, instance?.filteredRows);
+    }
+  }, [state, instance?.filteredRows, onFilter]);
```

Comment:
I don't think `instance` can be undefined. Why using `instance?.<....>`?

Reply:
This has been fixed. I was using it inside of `filterHandler.ts` but forgot to remove the check whenever code was moved to `Table.tsx`